packages/imperative/src/events/src/EventOperator.ts

```diff
@@ -0,0 +1,170 @@
+     * @static
+     * @returns {IZoweProcessor} The Zowe event processor instance.
+     * @throws {ImperativeError} If the application name is not recognized.
+     */
+    public static getZoweProcessor(): IZoweProcessor {
+        return this.createProcessor("Zowe", IProcessorTypes.BOTH, Logger.getAppLogger());
+    }
+
+    /**
+     * Retrieves a generic event processor that can emit and watch events.
+     *
+     * @static
+     * @param {string} appName - The application name.
+     * @param {Logger} [logger] - Optional logger for the processor.
+     * @returns {IEmitterAndWatcher} An event processor capable of both emitting and watching.
+     * @throws {ImperativeError} If the application name is not recognized.
+     */
+    public static getProcessor(appName: string, logger?: Logger): IEmitterAndWatcher {
+        return this.createProcessor(appName, IProcessorTypes.BOTH, logger);
+    }
+
+    /**
+     * Retrieves a watcher-only event processor.
+     *
+     * @static
+     * @param {string} appName - The application name, defaults to "Zowe" if not specified.
+     * @param {Logger} [logger] - Optional logger for the processor.
+     * @returns {IWatcher} A watcher-only event processor.
+     * @throws {ImperativeError} If the application name is not recognized.
+     */
+    public static getWatcher(appName: string = "Zowe", logger?: Logger): IWatcher {
+        return this.createProcessor(appName, IProcessorTypes.WATCHER, logger);
+    }
+
+    /**
+     * Retrieves an emitter-only event processor.
+     *
+     * @static
+     * @param {string} appName - The application name.
+     * @param {Logger} [logger] - Optional logger for the processor.
+     * @returns {IEmitter} An emitter-only event processor.
+     * @throws {ImperativeError} If the application name is not recognized.
+     */
+    public static getEmitter(appName: string, logger?: Logger): IEmitter {
+        return this.createProcessor(appName, IProcessorTypes.EMITTER, logger);
+    }
+
+    /**
+     * Deletes a specific type of event processor (emitter).
+     *
+     * @static
+     * @param {string} appName - The application name associated with the emitter to be deleted.
+     */
+    public static deleteEmitter(appName: string): void {
+        this.destroyProcessor(appName);
+    }
+
+    /**
+     * Deletes a specific type of event processor (watcher).
+     *
+     * @static
+     * @param {string} appName - The application name associated with the watcher to be deleted.
+     */
+    public static deleteWatcher(appName: string): void {
+        this.destroyProcessor(appName);
+    }
+
+    /**
+     * Deletes an event processor, removing both its emitter and watcher capabilities.
+     *
+     * @static
+     * @param {string} appName - The application name whose processor is to be deleted.
+     */
+    public static deleteProcessor(appName: string): void {
+        this.destroyProcessor(appName);
+    }
```

Comment:
These methods seem redundant since the implementation is identical. Could they be replaced with a single method `delete` or `deleteProcessor`?

Reply:
Hi so the reason we separated these out is so that watchers and emitters have consistent typedoc information. we wanted to keep consistency with how we were naming methods (tied to processor type) @zFernand0 @t1m0thyj